File: php/text_my_wife.php
#!/usr/bin/env php
<?php

require 'vendor/autoload.php';
(new Dotenv\Dotenv(__DIR__))->load();

(strpos(exec('who'), getenv('USER')) !== false) or exit('no session');

// Phone numbers
$my_number = '+xxx';
$her_number = '+xxx';

$reasons = [
	'Working hard',
	'Gotta ship this feature',
	'Someone fucked up the system again'
];

$rand = rand(0,count($reasons)-1);
$random_reason = $reasons[$rand];

$message = 'Late at work. '.$random_reason;

// Send a text message
$twilio = new Services_Twilio(getenv('TWILIO_ACCOUNT_SID'), getenv('TWILIO_AUTH_TOKEN'));
$twilio->account->messages->sendMessage(
        $my_number,
        $her_number,
        $message
);

echo 'Message sent at: #'.date('Y-m-d').' | Reason: '.$random_reason;

File: go/text_my_wife.go
package main

import (
	"fmt"
	"math/rand"
	"os"
	"os/exec"
	"strings"
	"time"
)

func main() {
	output1, err := exec.Command("who").Output()
	output2 := os.Getenv("USER")
	users := string(output1[:])
	current_user := string(output2[:])
	if !strings.Contains(users, current_user) {
		return
	}

	reasons := []string{"Working hard", "Gotta ship this feature", "Someone fucked the system again"}

	rand.Seed(time.Now().UTC().UnixNano())
	message := "Late at work. " + reasons[rand.Intn(len(reasons))]

	TWILIO_ACCOUNT_SID := string(os.Getenv("TWILIO_ACCOUNT_SID"))
	TWILIO_AUTH_TOKEN := string(os.Getenv("TWILIO_AUTH_TOKEN"))
	MY_NUMBER := string(os.Getenv("MY_NUMBER"))
	HER_NUMBER := string(os.Getenv("HER_NUMBER"))

	response, err := exec.Command("curl", "-fSs", "-u", TWILIO_ACCOUNT_SID+":"+TWILIO_AUTH_TOKEN, "-d", "From="+MY_NUMBER, "-d", "To="+HER_NUMBER, "-d", "Body="+message, "https://api.twilio.com/2010-04-01/Accounts/"+TWILIO_ACCOUNT_SID+"/Messages").Output()
	if err != nil {
		fmt.Printf("Failed to send SMS: %s", err)
		return
	}

	fmt.Printf("Message Sent Successfully with response: %s ", response)
}
